### PR TITLE
New: `prefer-arrow-callback` rule

### LINKED ISSUE #3140: Rule Suggestion: Prefer Arrow Callback
A rule for ES6 to recommend use of arrow functions in callback

Warnings when Arrow function is preferred
``` js
func( function () {...} )
setTimeout(function () {...}, 3000);
```

Not Warnings when Arrow function is preferred
```js 
func( () => {...} )
setTimeout( () => {...}, 3000);
```

Warnings when standard es5 function is preferred
```js 
func( () => {...} )
setTimeout( () => {...}, 3000);
```

Not Warnings when standard es5 function is preferred
``` js
func( function () {...} )
setTimeout(function () {...}, 3000);
```

Rule Example
``` js
"prefer-arrow-callback": [2, "always"]
```

or rule could alternatively look like:
``` js
"callback-style": [2, "arrow"]
```

ref conversation in #3113

### PR DESCRIPTION
**Pull request by mysticatea**
Fixes #3140.

### CODE REVIEW

File: docs/rules/prefer-arrow-callback.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Suggest using arrow functions as callbacks. (prefer-arrow-callback)
+
+Arrow functions are suited to callbacks, because:
+
+- Its `this` keyword binds to the upper scope's.
+- Its notation is shorter than function expressions.
```
Comment (nzakas) [inline docs/rules/prefer-arrow-callback.md]: "It's" for both lines, short for "it is"

File: docs/rules/prefer-arrow-callback.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+## Rule Details
+
+This rule is aimed to flag usage of function expressions in an argument list.
+
+The following patterns are considered warnings:
+
+```js
+foo(function(a) { return a; });
+```
+
```
Comment (nzakas) [inline docs/rules/prefer-arrow-callback.md]: Can you combine these code blocks?

File: docs/rules/prefer-arrow-callback.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+foo(function() { return this.a; }.bind(this));
+```
+
+The following patterns are not considered warnings:
+
+```js
+foo(a => a);
+```
+
+```js
```
Comment (nzakas) [inline docs/rules/prefer-arrow-callback.md]: Also combine these code blocks.

File: lib/rules/prefer-arrow-callback.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var stack = [];
+
+    /**
+     * Pushs `false` to the stack.
```
Comment (BYK) [inline lib/rules/prefer-arrow-callback.js]: I'd rather this explain the reason since it is quite obvious what the function does :)

Comment (xjamundx) [inline lib/rules/prefer-arrow-callback.js]: Also `pushes` right?

Comment (mysticatea) [inline lib/rules/prefer-arrow-callback.js]: Oh, sorry, I was half sleeping at that time...
That comment is useless completely.

**Comment by mysticatea** [comment]
Thank you for the review.

I modified those.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**